### PR TITLE
RSL1m4 If clientId is different, error from server

### LINKED ISSUE #101: The library should not be throwing an exception when a message is sent with different `clientId` from `auth.clientId`
[link to the check in question](https://github.com/ably/ably-php/blob/main/src/Channel.php#L103-L111).

As per [RSL1m4](https://docs.ably.io/client-lib-development-guide/features/#RSL1m4) this should instead result in the message being rejected by the server.

### PR DESCRIPTION
**Pull request by jdavid**
Fixes #101

Includes commits from PR #102

### CODE REVIEW

**Review by owenpearson — APPROVED** [review]
All good, but I'll wait for 102 to land before merging.

**Comment by owenpearson** [comment]
@jdavid looks like the error code will need to be updated here too, do you mind updating it and then I'll merge?

**Comment by jdavid** [comment]
@owenpearson Done, rebased over main, there's only 1 commit now.

**Pull request merged**
